assert/assertions_test.go

```diff
@@ -135,24 +135,34 @@ func TestObjectsAreEqual(t *testing.T) {
 
 		})
 	}
+}
 
-	// Cases where type differ but values are equal
-	if !ObjectsAreEqualValues(uint32(10), int32(10)) {
-		t.Error("ObjectsAreEqualValues should return true")
-	}
-	if ObjectsAreEqualValues(0, nil) {
-		t.Fail()
-	}
-	if ObjectsAreEqualValues(nil, 0) {
-		t.Fail()
-	}
+func TestObjectsAreEqualValues(t *testing.T) {
+	now := time.Now()
 
-	tm := time.Now()
-	tz := tm.In(time.Local)
-	if !ObjectsAreEqualValues(tm, tz) {
-		t.Error("ObjectsAreEqualValues should return true for time.Time objects with different time zones")
+	cases := []struct {
+		expected interface{}
```

Comment:
I would ask to replace `interface` with `any`

Reply:
Sure. I haven't really used generics that much. I keep forgetting that it exists :P

Reply:
We can't use generics yet. This project is still on go `v1.17`

https://github.com/stretchr/testify/blob/c3b0c9b4f50cd8320ccdcdfd3ffd6afc5b109c4a/go.mod#L5